velox/functions/prestosql/HyperLogLogFunctions.h

```diff
@@ -81,4 +81,84 @@ struct EmptyApproxSetWithMaxErrorFunction {
     return true;
   }
 };
+
+template <typename T>
+struct MergeHllFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<HyperLogLog>& result,
+      const arg_type<Array<HyperLogLog>>& input) {
+    using common::hll::DenseHll;
+    using common::hll::SparseHll;
+
+    // Create a memory pool and allocator for this call
+    auto pool = memory::memoryManager()->addLeafPool();
```

Comment:
We don't want to create a new memory pool and HSA for each row being processed.

CC: @xiaoxmeng @amitkdutta

Reply:
Updated this, please check new implementation